app/src/main/java/com/bashkevich/androidfundamentals/model/workmanager/MovieLoaderWorker.kt

```diff
@@ -38,29 +49,26 @@ class MovieLoaderWorker(appContext: Context, params: WorkerParameters) : Corouti
                     database.moviesDao.insertAllGenres(genreEntities)
                 }
 
-                launch {
-                    nowPlayingMoviesIds =
-                        RetrofitModule.moviesApi.getNowPlayingMovies().results.map { it.id }
+                nowPlayingMoviesIds = async {
+                    RetrofitModule.moviesApi.getNowPlayingMovies().results.map { it.id }
                 }
 
-                launch {
-                    upcomingMoviesIds =
-                        RetrofitModule.moviesApi.getUpcomingMovies().results.map { it.id }
+                upcomingMoviesIds = async {
+                    RetrofitModule.moviesApi.getUpcomingMovies().results.map { it.id }
                 }
 
-                launch {
-                    popularMoviesIds =
-                        RetrofitModule.moviesApi.getPopularMovies().results.map { it.id }
+                popularMoviesIds = async {
+                    RetrofitModule.moviesApi.getPopularMovies().results.map { it.id }
                 }
 
-                launch {
-                    topRatedMoviesIds =
-                        RetrofitModule.moviesApi.getTopRatedMovies().results.map { it.id }
+                topRatedMoviesIds = async {
+                    RetrofitModule.moviesApi.getTopRatedMovies().results.map { it.id }
                 }
             }
 
-            allMoviesIds = topRatedMoviesIds.plus(popularMoviesIds).plus(upcomingMoviesIds)
-                .plus(nowPlayingMoviesIds).distinct()
+            allMoviesIds = topRatedMoviesIds.await().plus(popularMoviesIds.await())
+                .plus(upcomingMoviesIds.await())
+                .plus(nowPlayingMoviesIds.await()).distinct()
 
             allMoviesIds.forEach { movieId ->
                 coroutineScope {
```

Comment:
Why have one more coroutine scope can you still work on that you use to load movies and get ids?